### PR TITLE
chore: add editorconfig

### PR DESCRIPTION
**Pull request by armano2**
This PR adds editorconfig file

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,9 @@
+root = true
+
+[*]
+charset = utf-8
+indent_style = space
+indent_size = 2
+end_of_line = lf
+insert_final_newline = true
+trim_trailing_whitespace = true
```
Comment (j-f1) [inline .editorconfig]: Can you add `quote_type = single` and delete `.prettierrc`?

**Comment by JamesHenry** [comment]
Is this for cases where an editor doesn't support a prettier extension?

**Comment by armano2** [comment]
yes, and editorconfig is used by github, github windows, and its build in to elot of editors
https://editorconfig.org/#download

**Comment by JamesHenry** [comment]
I don't really want to move away from prettier, I misunderstood this at first. When @j-f1 requested you remove .prettierrc I assumed that prettier must therefore support .editorconfig files as a source of config, but it doesn't seem like that's the case?

**Comment by armano2** [comment]
this is a case, prettierrc is using .editorconfig

**Comment by armano2** [comment]
and it used by default, you have to explicitly states to not use it in cli https://prettier.io/docs/en/cli.html#no-editorconfig

**Comment by j-f1** [comment]
@JamesHenry no that’s correct — Prettier does support loading an editorconfig. This PR doesn’t move away from Prettier, it just converts the config to a more portable format.

**Comment by JamesHenry** [comment]
Ok cool, thanks!

**Comment by JamesHenry** [comment]
I'm about to board a 7 hour flight just FYI, so might suddenly drop off the grid :)

**Pull request merged**